app/src/main/java/ch/epfl/sweng/erpa/activities/DiceActivity.java

```diff
@@ -1,152 +1,126 @@
+     * @param view not used
      */
-    private void rollAndShowAllDice(int[] rolls) {
-        int n = 0;
-        for (int i = 0; i < 7; ++i) {
-            String dieType = "D" + dice[i];
-            for (int j = 0; j < rolls[i]; ++j) {
-                allResultViews[n].setText(dieType + ": " + rollDie(dieType));
-                ++n;
+    @OnClick(R.id.rollButton)
+    public void rollDices(View view) {
+        if (Stream.of(allDice).allMatch(dice -> !dice.isRolling())) {
+            for (DieSketch die : allDice) {
+                die.roll();
             }
         }
     }
 
     /**
-     * Rolls one die
-     * @param dieType The type of die to roll
-     * @return The result of the roll
+     * Removes the last die on the layout
+     * @param view
      */
-    private int rollDie(String dieType) {
-        int result = -1;
-        switch (dieType) {
-            case "D4":
-                result = rng.nextInt(4);
-                break;
-            case "D6":
-                result = rng.nextInt(6);
-                break;
-            case "D8":
-                result = rng.nextInt(8);
-                break;
-            case "D10":
-                result = rng.nextInt(10);
-                break;
-            case "D12":
-                result = rng.nextInt(12);
-                break;
-            case "D20":
-                result = rng.nextInt(20);
-                break;
-            case "D100":
-                result = rng.nextInt(100);
-                break;
-            default:
-                throw new AssertionError("Die type is incorrect");
+    @OnClick(R.id.dice_layout)
+    public void removeDie(View view) {
+        if (!allDice.isEmpty()) {
+            int index = allDice.size() - 1;
+            allDice.remove(index);
+            flowLayout.removeViewAt(index);
         }
-        return ++result;
     }
 
     /**
-     * Gets all the TextViews where we write the results of the rolls
-     * @return An array containing all the TextViews
+     * Add a die on the FlowLayout if there is place depending on which button is pressed
+     * @param view The button that has been pressed, needs to be cast to Button
      */
-    private TextView[] getAllResultViews() {
-        TextView[] list = new TextView[MAX_DICE_NUMBER];
-        for (int i = 1; i < list.length + 1; ++i) {
-            String textId = "roll" + i;
-            int id = getResources().getIdentifier(textId, "id", getBaseContext().getPackageName());
-            list[i - 1] = findViewById(id);
+    public void addAndUpdateDie(View view) {
+        if (allDice.size() < MAX_DICE_NUMBER) {
+            Button button = (Button) view;
+            addAndShowDie(Integer.parseInt(button.getText().toString().substring(1)));
         }
-        return list;
     }
 
     /**
-     * Gets all the dice to be rolled
-     * @return An array containing all the number of rolls of each die
+     * Add a button of type dieType on the layout
```

Comment:
It seems like you add a DieSketch and not a button. I don't understand what you mean by "a button of type dieType".